packages/share/lib/share.dart

```diff
@@ -51,4 +52,128 @@ class Share {
 
     return channel.invokeMethod<void>('share', params);
   }
+
+  /// Summons the platform's share sheet to share a single file.
+  ///
+  /// Wraps the platform's native share dialog. Can share a file.
+  /// It uses the `ACTION_SEND` Intent on Android and `UIActivityViewController`
+  /// on iOS.
+  ///
+  /// The optional `sharePositionOrigin` parameter can be used to specify a global
+  /// origin rect for the share sheet to popover from on iPads. It has no effect
+  /// on non-iPads.
+  ///
+  /// May throw [PlatformException] or [FormatException]
+  /// from [MethodChannel].
+  static Future<void> shareFile(
+    File file, {
+    String mimeType,
+    String subject,
+    String text,
+    Rect sharePositionOrigin,
+  }) {
+    assert(file != null);
+    assert(file.existsSync());
+
+    return shareFiles(
+      <File>[file],
+      mimeTypes: <String>[mimeType ?? _mimeTypeForFile(file)],
+      subject: subject,
+      text: text,
+      sharePositionOrigin: sharePositionOrigin,
+    );
+  }
+
+  /// Summons the platform's share sheet to share multiple files.
+  ///
+  /// Wraps the platform's native share dialog. Can share a file.
+  /// It uses the `ACTION_SEND` Intent on Android and `UIActivityViewController`
+  /// on iOS.
+  ///
+  /// The optional `sharePositionOrigin` parameter can be used to specify a global
+  /// origin rect for the share sheet to popover from on iPads. It has no effect
+  /// on non-iPads.
+  ///
+  /// May throw [PlatformException] or [FormatException]
+  /// from [MethodChannel].
+  static Future<void> shareFiles(
+    List<File> files, {
+    List<String> mimeTypes,
+    String subject,
+    String text,
+    Rect sharePositionOrigin,
+  }) {
+    assert(files != null);
+    assert(files.isNotEmpty);
+    assert(files.every((file) => file.existsSync()));
+    final Map<String, dynamic> params = <String, dynamic>{
+      'paths': files.map((file) => file.path).toList(),
+      'mimeTypes':
+          mimeTypes ?? files.map((file) => _mimeTypeForFile(file)).toList(),
+    };
+
+    if (subject != null) params['subject'] = subject;
+    if (text != null) params['text'] = text;
+
+    if (sharePositionOrigin != null) {
+      params['originX'] = sharePositionOrigin.left;
+      params['originY'] = sharePositionOrigin.top;
+      params['originWidth'] = sharePositionOrigin.width;
+      params['originHeight'] = sharePositionOrigin.height;
+    }
+
+    return channel.invokeMethod('shareFiles', params);
+  }
+
+  static String _mimeTypeForFile(File file) {
```

Comment:
How about using: 

https://pub.dev/packages/mime

That way you don't have to maintain this list yourself.

Reply:
Good point. Switching to it now.